### PR TITLE
Moving ServiceFactory interfaces to internal.

### PR DESCRIPTION
**Pull request by tlm**
With this PR we are moving the ServiceFactory interfaces defined in the worker outside to an internal package. By doing this we can avoid the layer violation of having a bunch of Juju components referencing types out of the worker and subsequently getting the package imports from the worker.

This is needed for future PR's on dqlite that are introducing new testing suites for the ServiceFactory.

This change is self contained and only deals with the interfaces defined in the worker.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Not required.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-[4652]

### CODE REVIEW

File: worker/servicefactory/worker.go [hunk 1]
```diff
@@ -11,7 +11,8 @@ import (
 
 	"github.com/juju/juju/core/changestream"
 	coredatabase "github.com/juju/juju/core/database"
-	"github.com/juju/juju/domain/servicefactory"
+	domainsf "github.com/juju/juju/domain/servicefactory"
```
Comment (hpidcock) [inline worker/servicefactory/worker.go:14]: Be consistent. In another file you made this `domainservicefactory`, which I think is better.

**Review by hpidcock — APPROVED** [review]
Only one issue that should be fixed before landing.

File: worker/servicefactory/worker.go (hunk 1 above)
Comment (tlm) [reply worker/servicefactory/worker.go]: Done & Done

**Pull request merged**